backend/lib/src/data/rpc/client.rs

```diff
@@ -0,0 +1,228 @@
+            .await
+            .map_err(|e| Error::Rpc(jsonrpsee::core::client::Error::Custom(e.to_string())))?;
+
+        // Convert hex string to bytes
+        hash.ok_or_else(|| Error::NotFound("Block hash not found".to_string()))
+            .and_then(|h| {
+                hex::decode(h.trim_start_matches("0x")).map_err(|e| {
+                    Error::Rpc(jsonrpsee::core::client::Error::Custom(format!(
+                        "Invalid hex: {}",
+                        e
+                    )))
+                })
+            })
+    }
+
+    /// Submit a storage request transaction
+    pub async fn submit_storage_request(
+        &self,
+        location: Vec<u8>,
+        fingerprint: Vec<u8>,
+        size: u64,
+        peer_ids: Vec<Vec<u8>>,
+    ) -> Result<TransactionReceipt> {
+        // Create the storage request parameters
+        let params = json!({
+            "location": location,
+            "fingerprint": fingerprint,
+            "size": size,
+            "peer_ids": peer_ids,
+        });
+
+        // Submit the extrinsic
+        let tx_hash: String = self
+            .connection
+            .call("author_submitStorageRequest", params)
+            .await
+            .map_err(|e| Error::Rpc(jsonrpsee::core::client::Error::Custom(e.to_string())))?;
+
+        // Wait for transaction finalization and get receipt
+        let receipt_params = json!([tx_hash]);
+        let receipt: TransactionReceipt = self
+            .connection
+            .call("storagehub_getTransactionReceipt", receipt_params)
+            .await
+            .map_err(|e| Error::Rpc(jsonrpsee::core::client::Error::Custom(e.to_string())))?;
+
+        Ok(receipt)
+    }
+
+    /// Get storage request status
+    pub async fn get_storage_request_status(&self, file_key: &[u8]) -> Result<Option<String>> {
+        let params = json!([file_key]);
+
+        self.connection
+            .call("storagehub_getStorageRequestStatus", params)
+            .await
+            .map_err(|e| Error::Rpc(jsonrpsee::core::client::Error::Custom(e.to_string())))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::data::rpc::{AnyRpcConnection, ErrorMode, MockConnection};
+
+    #[tokio::test]
+    async fn test_get_file_metadata() {
+        // Create mock connection
+        let mock_conn = MockConnection::new();
+        mock_conn.set_response(
+            "storagehub_getFileMetadata",
+            json!({
+                "owner": [4, 5, 6],
+                "bucket_id": [7, 8, 9],
+                "location": [10, 11, 12],
+                "fingerprint": [13, 14, 15],
+                "size": 1024,
+                "peer_ids": [[16, 17], [18, 19]]
+            }),
+        );
+
+        let connection = Arc::new(AnyRpcConnection::Mock(mock_conn));
+        let client = StorageHubRpcClient::new(connection);
+
+        // Test the method
+        let result = client.get_file_metadata(&[1, 2, 3]).await.unwrap();
```

Comment:
I don't see where file_key = [1,2,3] is set on this test. It might be good to group all these tests values/structs in some constants section